RocketChat/Rocket.Chat
# fix: System messages are counted as agents' first responses in livechat rooms

## Proposed changes (including videos or screenshots)

 - Stop considering system messages (such as "agent joined room") as agents' first responses in livechat rooms. This caused the "average first response time" and "best first respone time" metrics to be much lower than expected.

## Issue(s)


## Steps to test or reproduce

Add a new agent to the woskspace, then create a livechat conversation and make sure to serve it to the new agent. Then, go to **Omnichannel > Analytics** and set "**Productivity**" as the analytics type.

**Current behavior:** the "Best first response time" and "Average of first response time" metrics will always be lower than the actual time that the agent took to respond to the livechat room (eg possibly 1s). System messages (such as "agent joined room") are being interpreted as first responses, which causes these metrics to go much lower;
**Expected behavior:** RC should only consider a first response in a room when the agent sends a text message. System messages should be disconsidered in the "Best first response time" and "Average of first response time" metrics.

## Further comments

[CORE-562](https://rocketchat.atlassian.net/browse/CORE-562)

[CORE-562]: https://rocketchat.atlassian.net/browse/CORE-562?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
